sdcm/nemesis.py

```diff
@@ -208,6 +208,22 @@ def disrupt_soft_reboot_node(self):
         self.log.info('Waiting JMX services to start after node reboot')
         self.target_node.wait_jmx_up()
 
+    def disrupt_restart_with_resharding(self):
+        self._set_current_disruption('RestartNode with resharding %s' % self.target_node)
+        murmur3_partitioner_ignore_msb_bits = 15
+        self.log.info('Restart node with resharding. New murmur3_partitioner_ignore_msb_bits value: '
+                      '{murmur3_partitioner_ignore_msb_bits}'.format(**locals()))
+        self.target_node.restart_node_with_resharding(murmur3_partitioner_ignore_msb_bits=murmur3_partitioner_ignore_msb_bits)
+        self.log.info('Waiting scylla services to start after node restart')
+        self.target_node.wait_db_up()
+        self.log.info('Waiting JMX services to start after node restart')
+        self.target_node.wait_jmx_up()
+
+        # Wait 5 minutes our before return back the default value
+        time.sleep(360)
+        self.log.info('Set back murmur3_partitioner_ignore_msb_bits value to 12')
```

Comment:
So we do two resharding each time?
How long each resharding takes for 1TB?
can't we reshard with a random choice of number (even out of list) and leave it as is?

Reply:
> So we do two resharding each time?
I added this by your definition
> How long each resharding takes for 1TB?
I didn't check this
> can't we reshard with a random choice of number (even out of list) and leave it as is?